src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/services/service-form/service-form.component.ts

```diff
@@ -244,6 +258,79 @@ export class ServiceFormComponent extends CdForm implements OnInit {
     this.cephServiceService.list().subscribe((services: CephServiceSpec[]) => {
       this.services = services.filter((service: any) => service.service_type === 'rgw');
     });
+
+    if (this.editing) {
+      this.action = this.actionLabels.EDIT;
+      this.disableForEditing(this.serviceType);
+      this.cephServiceService.list(this.serviceName).subscribe((response: CephServiceSpec[]) => {
+        const formKeys = ['service_type', 'service_id', 'unmanaged'];
+        formKeys.forEach((keys) => {
```

Comment:
Sorry for my ignorance with this part of the code, but I'm a bit astonished about how complicate is to populate the form...  We have lots of back-end data harcoded here (`service_type`, `service_id`, etc). Couldn't we get this info from the back-end? Can't we use a UI API endpoint to provide all this information already processed from the back-end, so that the front-end only has to "render" it?

Reply:
I think its possible. I was trying it out locally but so far not successful, but I'd like to try it out. I've created a [tracker](https://tracker.ceph.com/issues/53268) and I'll work on it in separate PR because the changes will be out of scope of this PR and will be messier. I hope its okay @epuertat